changelog/9877.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed ``caplog.get_records(when)`` still get the stage records after ``caplog.clear()``
```

Comment:
```suggestion
ensure ``caplog.get_records(when)`` returns current/correct data after invoking ``caplog.clear()``
```
as maybe a nitpick, will merge in the evening

Reply:
thanks for the feedback!